### PR TITLE
Add blocktests as go tests

### PR DESCRIPTION
**Pull request by Gustav-Simonsson**
* Merge of Felix's fixes to geth blocktest cmd and HEX encodings
* Add fixes to parsing and converting of fields in tx tests
* Correct logic in tx tests; validation of fields and correct
  logic for when RLP decoding works/fails and when this is
  expected or not
* Rename files for consistency
* Add block tests wrapper to run block tests with go test

### CODE REVIEW

File: core/chain_manager.go [hunk 1]
```diff
@@ -92,6 +92,7 @@ type ChainManager struct {
 	futureBlocks *BlockCache
 
 	quit chan struct{}
+	wg   sync.WaitGroup
```
Comment (obscuren) [inline core/chain_manager.go]: First of all, these sort of changes need to be well documented. Second I suggest you find some other way to solve this.

Comment (fjl) [inline core/chain_manager.go]: I made these changes while working out a crash in the tests together with Gustav. The blockpool was crashing because it didn't expect to be started and stopped in short succession.

While looking at the source of the events, I found that `ChainManager` now has a background goroutine which is never stopped. The changes ensure that it is stopped, and also that stopping blocks until the loop is down.

> Second I suggest you find some other way to solve this.

`sync.WaitGroup` is the canonical way to wait for one or more goroutines to end. But it could also be done with a channel instead, if you prefer that.

Comment (fjl) [inline core/chain_manager.go]: Either way, the issue doesn't really exist anymore because the blockpool is no longer used. It would still be preferable to fix `ChainManager` so it exits properly.

@Gustav-Simonsson maybe you could strip those commits (changes to blockpool and the chain manager) from the PR. I'll submit a separate PR with the change to core.

Comment (obscuren) [inline core/chain_manager.go]: The fix is fine with a little context :-)

Comment (Gustav-Simonsson) [inline core/chain_manager.go]: Sure thing, will cherry pick the commits from @fjl branch then to leave this out for a separate PR.

**Comment by Gustav-Simonsson** [comment]
Closing PR as a new one will be opened excluding some commits.
